management/virtualmachine/client.go

```diff
@@ -505,19 +504,18 @@ func (self VirtualMachineClient) addDockerPort(configurationSets []Configuration
 		}
 
 		dockerEndpoint := self.createEndpoint("docker", "tcp", dockerPort, dockerPort)
-		configurationSets[i].InputEndpoints.InputEndpoint = append(configurationSets[i].InputEndpoints.InputEndpoint, dockerEndpoint)
+		*configurationSets[i].InputEndpoints = append(*configurationSets[i].InputEndpoints, dockerEndpoint)
 	}
 
 	return nil
 }
 
-func (self VirtualMachineClient) createVMDeploymentConfig(role *Role) VMDeployment {
-	deployment := VMDeployment{}
+func (self VirtualMachineClient) createVMDeploymentConfig(role *Role) Deployment {
+	deployment := Deployment{}
 	deployment.Name = role.RoleName
-	deployment.Xmlns = azureXmlns
 	deployment.DeploymentSlot = "Production"
 	deployment.Label = role.RoleName
-	deployment.RoleList.Role = append(deployment.RoleList.Role, role)
+	deployment.RoleList = append(deployment.RoleList, role)
```

Comment:
doesn't this append look a bit weird to you?

also we should probably use struct initializer here.